### PR TITLE
[fir] Sync back after D112845

### PR DESCRIPTION
**Pull request by clementval**
Sync back after small change made when upstreaming https://reviews.llvm.org/D112845

### CODE REVIEW

File: flang/lib/Optimizer/CodeGen/CodeGen.cpp [hunk 1]
```diff
@@ -2520,6 +2525,8 @@ struct GlobalOpConversion : public FIROpConversion<fir::GlobalOp> {
     return true;
   }
 
+  // TODO: String comparaison should be avoided. Replace linkName with an
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/CodeGen/CodeGen.cpp:2528]: Typo: comparison.

File: flang/include/flang/Optimizer/CodeGen/CGPasses.td [hunk 2]
```diff
@@ -23,10 +23,7 @@ def FIRToLLVMLowering : Pass<"fir-to-llvm-ir", "mlir::ModuleOp"> {
     will also convert ops in the standard and FIRCG dialects.
   }];
   let constructor = "::fir::createFIRToLLVMPass()";
-  let dependentDialects = [
-    "fir::FIROpsDialect", "fir::FIRCodeGenDialect", "mlir::BuiltinDialect",
-    "mlir::LLVM::LLVMDialect", "mlir::omp::OpenMPDialect"
-  ];
+  let dependentDialects = ["mlir::LLVM::LLVMDialect"];
```
Comment (schweitzpgi) [inline flang/include/flang/Optimizer/CodeGen/CGPasses.td:26]: Glad this got sorted out.

**Review by schweitzpgi — APPROVED** [review]

File: flang/include/flang/Optimizer/CodeGen/CGPasses.td (hunk 2 above)
Comment (clementval) [reply flang/include/flang/Optimizer/CodeGen/CGPasses.td:26]: Apparently it works. This was one of the comment that came frequently while upstreaming passes.

File: flang/lib/Optimizer/CodeGen/CodeGen.cpp (hunk 1 above)
Comment (clementval) [inline flang/lib/Optimizer/CodeGen/CodeGen.cpp:2528]:
```suggestion
  // TODO: String comparison should be avoided. Replace linkName with an
```

**Pull request merged**